PSeries SR-IOV Phase 1, Patch1

jjalvare opened the pull request:
Patch after comments from Bjorn Helgaas. This patch uses match_driver state to -1 to not load Virtual Function driver found in patch: https://patchwork.kernel.org/patch/9882915/

jjalvare committed:
pseries: Add PSeries SR-IOV Machine dependant calls

Add calls for PSeries platform to configure/deconfigure
SR-IOV.

jjalvare committed:
pseries/kernel: Do not match Virtual Function Driver

Set  VF pci device match_driver attribute to -1 state
thus attaching of driver will not happen when VF
device is dynamically loaded after configuring SR-IOV
in adapter card.

jjalvare committed:
powerpc/kernel: Add kernel object events when errors reported

Devices may go offline when EEH  has been reported. When
devices are shutdown KOBJ_OFFLINE is set for kernel
objects. When device resumes KOBJ_ONLINE is set. Therefore,
EEH events are better propagated to user space for devices.

bgly commented:
Prob drop this and split out the specific sriov_enable/disable stuff when we get more concrete rtas call stuff.

jjalvare committed:
powerpc/kernel: PSeries update VF config space after EEH

Add EEH platform operations for pseries to update VF
config space. With this change after EEH, the VF will have updated
config space for pseries platform.

jjalvare committed:
pseries/kernel: Pseries SR-IOV VF will remain assigned

In Pseries platforms the VFs created when configuring
SR-IOV will always remain assigned until removed.

jjalvare committed:
pseries/kernel Add EEH operations to notify resume

When Pseries SR-IOV is enabled and after a PF driver
has resumed from EEH, platform has to be notified
of the event so the child VFs can be allowed to
resume there normal recovery path.

This patch adds the eeh operation to platform
dependent code and allow unfreeze call to pseries
eeh platform.

bgly commented:
Can you close this if you aren't using it?
